### PR TITLE
adds lolin s3 pro

### PR DESCRIPTION
**Pull request by bsdlp**
took hwid from https://github.com/espressif/arduino-esp32/blob/c21a8cdaa56822b13e6f8259ca5b5b1f7b5d88ae/variants/lolin_s3_pro/pins_arduino.h#L8

### CODE REVIEW

File: boards/lolin_s3_pro.json [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+      "ldscript": "esp32s3_out.ld",
+      "partitions": "default_16MB.csv",
+      "memory_type": "qio_opi"
+    },
+    "core": "esp32",
+    "extra_flags": [
+      "-DARDUINO_LOLIN_S3_PRO",
+      "-DBOARD_HAS_PSRAM",
+      "-DARDUINO_USB_MODE=1",
+      "-DARDUINO_USB_CDC_ON_BOOT=1",
```
Comment (valeros) [inline boards/lolin_s3_pro.json:13]: Are you sure the `-DARDUINO_USB_CDC_ON_BOOT=1` macro is necessary? In the boards.txt it's [disabled](https://github.com/espressif/arduino-esp32/blob/master/boards.txt#L8140) by default.

File: boards/lolin_s3_pro.json [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    "f_flash": "80000000L",
+    "flash_mode": "qio",
+    "hwids": [
+      [
+        "0x303A",
+        "0x8161"
+      ]
+    ],
+    "mcu": "esp32s3",
+    "variant": "lolin_s3"
```
Comment (valeros) [inline boards/lolin_s3_pro.json:27]: Probably should be `lolin_s3_pro`.

File: boards/lolin_s3_pro.json [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  "upload": {
+    "flash_size": "16MB",
+    "maximum_ram_size": 327680,
+    "maximum_size": 16777216,
+    "require_upload_port": true,
+    "speed": 460800
+  },
+  "url": "https://www.wemos.cc/en/latest/s3/s3_pro.html",
+  "vendor": "WEMOS"
+}
```
Comment (valeros) [inline boards/lolin_s3_pro.json:50]: EOF is missing.

**Review by valeros — CHANGES_REQUESTED** [review]

**Comment by valeros** [comment]
Hi @bsdlp, thanks for the PR, please review comments below.

**Comment by bsdlp** [comment]
thanks @valeros i've addressed your comments

**Review by valeros — APPROVED** [review]

**Comment by valeros** [comment]
Thanks, merged.

**Pull request merged**